File: Data Structures & Algorithms/asteroid-collision/submission-0.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        stack = []
        for i in range(len(asteroids)):
            while stack and asteroids[i] < 0 < stack[-1]:
                diff = asteroids[i] + stack[-1]
                if diff < 0:
                    stack.pop()
                    continue
                elif diff == 0:
                    stack.pop()
                break
            else:
                stack.append(asteroids[i])
        return stack

File: Data Structures & Algorithms/daily-temperatures/submission-0.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        let arr = temperatures
        let res = Array(arr.length).fill(0);
        let stack = []
        for (let i = 0; i < arr.length; i++) {
            const t = arr[i]
            while(stack.length > 0 && t > stack[stack.length-1][0]){
                const [stackT, stackInd] = stack.pop()
                res[stackInd] = i - stackInd
            }
            stack.push([t, i])
        }
        return res
    }
}
